### PR TITLE
Fix "Nix" step to use `nix upgrade-nix` in more situations

### PR DESCRIPTION
**Pull request by 9999years**
`nix upgrade-nix` can be used on any platform except NixOS where `nix` is available.

Also use `nix profile upgrade --verbose` because the non-verbose mode doesn't print anything on stdout.

Also add `--extra-experimental-features nix-command` to `nix` invocations so that they don't error.

## Standards checklist:

- [x] The PR title is descriptive.
- [x] I have read `CONTRIBUTING.md`
- [x] The code compiles (`cargo build`)
- [x] The code passes rustfmt (`cargo fmt`)
- [x] The code passes clippy (`cargo clippy`)
- [x] The code passes tests (`cargo test`)
- [x] *Optional:* I have tested the code myself

### CODE REVIEW

File: src/steps/generic.rs [hunk 1]
```diff
@@ -417,7 +417,7 @@ pub fn run_pip3_update(ctx: &ExecutionContext) -> Result<()> {
     Command::new(&python3)
         .args(["-m", "pip"])
         .output_checked_utf8()
-        .map_err(|_| SkipStep("pip does not exists".to_string()))?;
+        .map_err(|_| SkipStep("pip does not exist".to_string()))?;
```
Comment (9999years) [inline src/steps/generic.rs:420]: Drive-by typo fixes

File: src/steps/os/unix.rs [hunk 2]
```diff
@@ -363,24 +365,23 @@ pub fn run_nix(ctx: &ExecutionContext) -> Result<()> {
...
+    #[allow(unused_mut)]
+    let mut should_self_upgrade = cfg!(target_os = "macos");
 
     #[cfg(target_os = "linux")]
     {
-        use super::linux::Distribution;
-
+        // We can't use `nix upgrade-nix` on NixOS.
         if let Ok(Distribution::NixOS) = Distribution::detect() {
-            return Err(SkipStep(String::from("Nix on NixOS must be upgraded via nixos-rebuild switch")).into());
```
Comment (9999years) [inline src/steps/os/unix.rs:380]: We shouldn't skip the whole step on NixOS, because we can still upgrade user-level packages with `nix-env --upgrade` or `nix profile upgrade '.*'`

**Review by SteveLauC — APPROVED** [review]

**Comment by SteveLauC** [comment]
LGTM, thanks! :)
